discovery-infra/tests/test_validations.py

```diff
@@ -80,3 +81,30 @@ def test_host_insufficient(self, cluster, modified_nodes):
         new_cluster.wait_until_hosts_are_discovered(allow_insufficient=True)
         new_cluster.wait_for_host_validation(vcpu_host_id, "hardware", "has-min-cpu-cores", ["success"])
         new_cluster.wait_for_host_validation(ram_host_id, "hardware", "has-min-memory", ["success"])
+
+    @pytest.mark.regression
+    @pytest.mark.skip(reason="BZ:1897916")
+    def test_hosts_and_cluster_max_length(self, cluster, nodes):
+        # Trying to create a cluster with more then 54 chars
+        cluster_name = "this-is-a-very-long-cluster-name-this-is-a-very-long-cluster-name-this" \
+                       "-is-a-very-long-cluster-name"
+        # TODO: assert on error code and reason
+        with pytest.raises(ApiException):
+            cluster(cluster_name)
+        c = cluster()
+        # Trying to update cluster name to have more then 54 chars
+        with pytest.raises(ApiException):
+            c.set_cluster_name(cluster_name)
+        # Set cluster name with exactly 54 chars
+        cluster_name = "this-is-exactly-54-chars-cluster-name-for-testing-abc"
+        c.set_cluster_name(cluster_name)
+        c.prepare_for_install(nodes=nodes)
+        random_master_host = c.get_hosts_by_role(role=consts.NodeRoles.MASTER)[0]
```

Comment:
lets add it to hosts. there is already get random host. lets have get random host by role